nix/modules/formatting.nix

```diff
@@ -1,16 +1,14 @@
 { inputs, ... }:
 {
   imports = [
-    inputs.treefmt-nix.flakeModule
+    (inputs.git-hooks + /flake-module.nix)
   ];
   perSystem = { config, self', pkgs, lib, ... }: {
-    # Add your auto-formatters here.
-    # cf. https://nixos.asia/en/treefmt
-    treefmt.config = {
-      projectRootFile = "flake.nix";
-      programs = {
+    pre-commit.settings = {
+      hooks = {
         nixpkgs-fmt.enable = true;
         rustfmt.enable = true;
+        cargo-check.enable = true;
```

Comment:
Just curious: does `cargo-check` add significant overhead during `git commit`?

Reply:
Didn't think this through. It should be somewhat slow.